src/main/java/com/mongodb/spark/sql/connector/MongoCatalog.java

```diff
@@ -0,0 +1,422 @@
+   * @throws IllegalArgumentException If any change is rejected by the implementation.
+   */
+  @Override
+  public Table alterTable(final Identifier identifier, final TableChange... changes)
+      throws NoSuchTableException {
+    if (!tableExists(identifier)) {
+      throw new NoSuchTableException(identifier);
+    }
+    throw new IllegalArgumentException("Altering collections is not supported.");
+  }
+
+  /**
+   * Drop a table (collection) in the catalog.
+   *
+   * <p>If the catalog supports views and contains a view for the identifier and not a table, this
+   * must not drop the view and must return false.
+   *
+   * @param identifier a table (collection) identifier
+   * @return true if a table (collection) was deleted, false if no table (collection) exists for the
+   *     identifier
+   */
+  @Override
+  public boolean dropTable(final Identifier identifier) {
+    assertInitialized();
+    if (identifier.namespace().length == 1
+        && Arrays.stream(listTables(identifier.namespace()))
+            .anyMatch(i -> Objects.equals(i.name(), identifier.name()))) {
+      mongoConnectionProvider.doWithCollection(
+          identifier.namespace()[0], identifier.name(), MongoCollection::drop);
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Renames a table (collection) in the catalog.
+   *
+   * <p>If the catalog supports views and contains a view for the old identifier and not a table,
+   * this throws {@link NoSuchTableException}. Additionally, if the new identifier is a table
+   * (collection) or a view, this throws {@link TableAlreadyExistsException}.
+   *
+   * <p>If the catalog does not support table (collection) renames between namespaces, it throws
+   * {@link UnsupportedOperationException}.
+   *
+   * @param oldIdentifier the table (collection) identifier of the existing table (collection) to
+   *     rename
+   * @param newIdentifier the new table (collection) identifier of the table
+   * @throws NoSuchTableException If the table (collection) to rename doesn't exist or is a view
+   * @throws TableAlreadyExistsException If the new table (collection) name already exists or is a
+   *     view
+   */
+  @Override
+  public void renameTable(final Identifier oldIdentifier, final Identifier newIdentifier)
+      throws NoSuchTableException, TableAlreadyExistsException {
+    if (!tableExists(oldIdentifier)) {
+      throw new NoSuchTableException(oldIdentifier);
+    } else if (tableExists(newIdentifier)) {
+      throw new TableAlreadyExistsException(newIdentifier);
+    }
+    mongoConnectionProvider.doWithCollection(
+        oldIdentifier.namespace()[0],
+        oldIdentifier.name(),
+        coll ->
+            coll.renameCollection(
+                new MongoNamespace(newIdentifier.namespace()[0], newIdentifier.name())));
+  }
+
+  private void assertInitialized() {
+    assert initialized : "The MongoCatalog is has not been initialized";
```

Comment:
In the driver we decided to thrown AssertionError rather than rely on the application to enable Java assertions (which are disabled by default).  See com.mongodb.assertions.Assertions#assertTrue

Reply:
Ack

Reply:
Ack